src/components/proofs/ERC721ProofSection.tsx

```diff
@@ -1,73 +1,54 @@
-import { BodyText, HintText } from 'components/Text'
+import { BodyText, HintText } from 'components/ui/Text'
+import { DataKeys } from 'models/DataKeys'
 import { Suspense } from 'preact/compat'
 import { useSnapshot } from 'valtio'
-import AvailableProofsList from 'components/proofs/AvailableProofsList'
 import HintCard from 'components/badges/HintCard'
-import Network from 'models/Network'
-import ProofSection from 'components/ProofSection'
+import Proof from 'models/Proof'
+import ProofSection from 'components/proofs/ProofSection'
 import ProofStore from 'stores/ProofStore'
-import ReadyERC721ProofsList from 'components/proofs/ReadyERC721ProofsList'
+import ProofsList from 'components/proofs/ProofsList'
+import data from 'data'
 import useProofAddressesAvailableToCreate from 'hooks/useProofAddressesAvailableToCreate'
 
-export function ERC721ProofSection({
-  account,
-  network,
-}: {
-  account: string
-  network: Network
-}) {
-  const { eRC721ProofsCompleted } = useSnapshot(ProofStore)
+export function ERC721ProofSection({ dataKey }: { dataKey: DataKeys }) {
+  const network = data[dataKey].network
+  const { proofsCompleted } = useSnapshot(ProofStore[dataKey])
   const networkProofAddressesAvailableToCreate =
     useProofAddressesAvailableToCreate(network)
 
-  const networkProofsCompleted = eRC721ProofsCompleted.filter(
-    (proof) => proof.network === network
-  )
-
-  const nothingToGenerateMainnet =
-    networkProofsCompleted.length === 0 &&
-    networkProofAddressesAvailableToCreate.length === 0
+  async function onCreate(original: string) {
+    await data[dataKey].createProof(ProofStore[dataKey], original)
+  }
 
   return (
-    <ProofSection title={<BodyText>{network} NFTs</BodyText>}>
-      <ReadyERC721ProofsList network={network} />
-      {account && (
-        <AvailableProofsList
-          proofs={networkProofAddressesAvailableToCreate}
-          network={network}
-        />
-      )}
-      {nothingToGenerateMainnet && (
-        <HintCard small marginY={false}>
-          <HintText bold center>
-            No NFTs to proof
-          </HintText>
-        </HintCard>
-      )}
-    </ProofSection>
+    <ProofsList
+      proofs={proofsCompleted as Proof[]}
+      originals={networkProofAddressesAvailableToCreate}
+      onCreate={onCreate}
+      dataKey={dataKey}
+      nothingToGenerateText="No NFTs to proof"
+    />
   )
 }
 
 export default function ERC721ProofSectionSuspended({
-  account,
-  network,
+  dataKey,
 }: {
-  account: string
-  network: Network
+  dataKey: DataKeys
 }) {
   return (
-    <Suspense
-      fallback={
-        <ProofSection title={<BodyText>{network} NFTs</BodyText>}>
+    <ProofSection title={<BodyText>{data[dataKey].proofTitle}</BodyText>}>
```

Comment:
Looks like you're getting this `data[dataKey]` thing many times, maybe it's better to get it only once somewhere here and pass the result into the components below?